### ISSUE TITLE
strange proxy behaviour

### ISSUE THREAD

**Issue by remkoboschker** [mitchlloyd/ember-autosave#24]
I hope I can communicate clearly what is going on. 

I have the autosave proxy on an input with an initial value. I type, I wait, I saves. Great.

If I go backspace to the initial value and wait. I does not save, not does it change the value in emberdata as viewed in the ember chrome extension (I also printed it to the console, to make sure). I noticed that the `setUnknownProperty` function is never called. 

This might be the standard behaviour, but it catches me out, because the input show a value (the initial value) and the last modified value is actually saved. 

It this as expected? Could you explain or point to a work-around? Thank you.

I can reproduce the behaviour changing the title of this repo's example app.

**Comment by remkoboschker** [comment]
Could be related https://github.com/emberjs/ember.js/issues/12621

**Comment by remkoboschker** [comment]
Cannot reproduce by setting values directly
````
test('setting a property eventually saves the model with the property reset to initial propery', function(assert) {
  var Component = Ember.Object.extend({
    autosaveObject: autosave('model')
  });

  model.name = 'Millie';

  var component = Component.create({ model: model });

  assert.equal(get(component, 'autosaveObject.name'), 'Millie');

  set(component, 'autosaveObject.name', 'Milliee');

  assert.equal(get(component, 'autosaveObject.name'), 'Milliee');

  assert.ok(!model.save.called, 'save was not called immediately');
  clock.tick(1000);
  assert.ok(model.save.called, 'save was called after ellapsed time');

  assert.equal(get(component, 'autosaveObject.name'), 'Milliee');

  set(component, 'autosaveObject.name', 'Millie');

  assert.ok(model.save.calledOnce, 'save was not called immediatly the second time');
  clock.tick(1000);
  assert.ok(model.save.calledTwice, 'save was called the second time after ellapsed time');
  assert.equal(get(component, 'autosaveObject.name'), 'Millie');
});
````
Perhaps to do with the binding of the input

**Comment by remkoboschker** [comment]
I managed to write a failing acceptance test for it 
https://github.com/curit/ember-autosave/blob/master/tests/acceptance/change-save-revert-test.js
I did change the mock backend to mirage

**Comment by mitchlloyd** [comment]
This was a tricky one! The proxy object needed to notify Ember whenever `setUnknownProperty` was called. Otherwise Ember's templating system would never know that things were changing. When setting the value back to what it was, the input helper would not call `set` because the cached value was equal to the new value.

Thanks for tracking this down and reproducing. Released fix as 2.0.1.
